app/src/main/java/com/example/myhouse/model/repository/RepositoryImpl.kt

```diff
@@ -27,14 +28,24 @@ class RepositoryImpl(
 //        backgroundThreadRealm.close()
 //    }
 
-    override fun getDoorsFromServer(): List<DoorDTO> {
-        val dto = remoteDataSource.api.getDoorsSource().execute().body()
-        return dto?.data!!
+    override fun getDoorsFromServer(): List<DoorDTO>? {
+        return try {
+            val dto = remoteDataSource.api.getDoorsSource().execute().body()
+            dto?.data
```

Comment:
В данном случае мы уже уверенны, что объект существует, иначе здесь же выдать ошибку

```suggestion
            dto!!.data
```

И вообще, лучше иметь больше проверок, в том числе и проверять содержимое.

Пример функции проверки в базовом классе viewModel

```kotlin
fun <T : BaseResponseModel> checkError(response: Response<T>, showMessage: Boolean = true) : Boolean {
        return if(!response.isSuccessful || response.body() == null)
            false
        else if(response.body()!!.status== ResponseStatus.success)
            true
        else {
            response.body()!!.errorMessage?.let {
                if(showMessage)
                    showErrorMessage(it)
            }
            false
        }
    }
```